### PR TITLE
Update 20200125

### PR DESCRIPTION
**Pull request by dwayne-hart**
Removal of GIOPS Polar Stereographic data sets;

       * GIOPS 10 day Forecast 3D - Polar Stereographic
       * GIOPS 10 day Forecast Surface - Polar Stereographic
       * GIOPS 10 day Forecast Surface - Polar Stereographic

Reordered the remaining data sets.

Updated following data sets to use the new *bathymetry_file_url* variable set to "/data/grids/ecc/misc/lat_lon/bathy_meter_giops_latlon_0.2deg_reformatted_lonlatupdate.nc"

       * GIOPS 10 day Forecast 3D - LatLon
       * GIOPS Daily Forecast 3D - LatLon

Added the new variable *depthexcess* to the following data sets;

       * GIOPS Daily Forecast 3D - LatLon
       * RIOPS Forecast 3D - Polar Stereographic

Removed some data sets which had the string "enabled": false defined.

### CODE REVIEW

**Review by nsoontie — COMMENTED** [review]
@dwayne-hart Can you please add the Depth Excess variable to dataset 6. RIOPS Forecast 3D - Polar Stereographic.

There is a comment in this pull request that describes the path to the bathymety_file_url for this dataset.
https://github.com/DFO-Ocean-Navigator/Ocean-Data-Map-Project/pull/753

**Comment by dwayne-hart** [comment]
@nsoontie, @VanessaSutton-Pande, and @htmlboss when the **06. RIOPS Forecast 3D - Polar Stereographic** data set is selected the temperature plot is rendered fairly quickly. However, when the *Depth Excess* variable is selected it took around 5 minutes to get the start of a plot and I stopped my performance analysis after 12 minutes of waiting. I'm not sure if this should be used in the public facing site. 

image

File: datasetconfig.json [hunk 1]
```diff
@@ -328,15 +185,16 @@
...
 	    "nitrogensaturation": { "hide": "true", "name": "Nitrogen Saturation", "scale": [8, 20], "unit": "ml/l", "equation": "nitrogensaturation(votemper - 273.15, vosaline)", "dims": ["time", "depth", "yc", "xc"] },
             "soniclayerdepth": { "name": "Sonic Layer Depth", "unit": "m", "scale": [1.5, 60], "equation": "soniclayerdepth([depth], latitude, [votemper] - 273.15, [vosaline])", "dims": ["time", "yc", "xc"] },
-            "deepsoundchannelbottom": { "name": "Critical Depth", "unit": "m", "scale": [500, 2000], "equation": "deepsoundchannelbottom([depth], latitude, [votemper] - 273.15, [vosaline])", "dims": ["time", "yc", "xc"] }
+            "deepsoundchannelbottom": { "name": "Critical Depth", "unit": "m", "scale": [500, 2000], "equation": "deepsoundchannelbottom([depth], latitude, [votemper] - 273.15, [vosaline])", "dims": ["time", "yc", "xc"] },
+	    "depthexcess": { "name": "Depth Excess", "unit": "m", "scale": [0, 1000], "equation": "depthexcess([depth], latitude, [votemper] - 273.15, [vosaline], bathy)", "dims": ["time", "latitude", "longitude"] }
```
Comment (nsoontie) [inline datasetconfig.json]: dims should be ["time", "yc", "xc"]

**Comment by htmlboss** [comment]
yikes. Yah that shouldn't be public. One of the dependent acoustic variable may be slowing things down since this one is just a basic subtraction statement.

File: datasetconfig.json [hunk 2]
```diff
@@ -302,11 +158,12 @@
         "envtype": ["ocean"],
         "enabled": true,
         "help": " Document <a href=\"../data-help/riops-3d.html\" target=\"_new\">link</a>.",
-        "name": "09. RIOPS Forecast 3D - Polar Stereographic",
+        "name": "06. RIOPS Forecast 3D - Polar Stereographic",
         "quantum": "hour",
         "type": "forecast",
         "cache": 6,
         "time_dim_units": "seconds since 1950-01-01 00:00:00",
+	"bathymetry_file_url": "/data/grids/ecc/misc/polar_stereographic/bathy_meter_giops_ps5km.nc",
```
Comment (nsoontie) [inline datasetconfig.json]: This file is for giops, not riops!! As referenced [here](https://github.com/DFO-Ocean-Navigator/Ocean-Data-Map-Project/pull/753#pullrequestreview-541260330), please use this file:
hank:/tank/data/grids/ecc/riops/bathy_meter_riops_ps5km_reformatted-ONav.nc

File: datasetconfig.json [hunk 3]
```diff
@@ -171,50 +98,11 @@
...
-            "magicevel": { "name": "Sea Ice Velocity", "envtype": "ice", "unit": "m/s", "scale": [0, 1], "equation": "magnitude(itzocrtx, itmecrty)", "dims": ["time", "yc", "xc"], "east_vector_component": "itzocrte", "north_vector_component": "itmecrtn" },
-            "iicestrength": { "name": "Compressive Ice Strength", "envtype": "ice", "unit": "Nm-1", "scale": [-1, 1] },
-            "iicepressure": { "name": "Internal Ice Pressure", "envtype": "ice", "unit": "Nm-1", "scale": [-1, 1] }
-        }
-    },
     "giops_fc_3dll": {
         "envtype": ["ocean", "ice"],
         "url": "/data/db/giops-fc3dll.sqlite3",
-        "name": "07. GIOPS Daily Forecast 3D - LatLon",
+	"bathymetry_file_url": "/data/grids/ecc/misc/lat_lon/bathy_meter_giops_latlon_0.2deg_reformatted_lonlatupdate.nc",
```
Comment (nsoontie) [inline datasetconfig.json]: For consistency with the [pull request](https://github.com/DFO-Ocean-Navigator/Ocean-Data-Map-Project/pull/753#pullrequestreview-541260330) associated with this change, please use this file:
hank:/tank/data/grids/ecc/giops/bathy_meter_giops_latlon_reformatted-ONav.nc

Also, the bathymetry_file_url in dataset 01. should reference this file.

**Comment by nsoontie** [comment]
> yikes. Yah that shouldn't be public. One of the dependent acoustic variable may be slowing things down since this one is just a basic subtraction statement.

There are a number of issues with Dwayne's update to the datasetconfig. This problem is likely related to using the incorrect bathymetry file. I've made comments with suggested revisions.

**Comment by htmlboss** [comment]
ok well in that case I'll wait until we have updated results.

**Comment by nsoontie** [comment]
Looks like my review was on a different commit and @dwayne-hart may have already resolved those issues. @dwayne-hart can you confirm whether or not the corrections to the riops bathymetry file/dims resolved the timing issue?

**Comment by dwayne-hart** [comment]
@nsoontie the corrections were made prior to testing the *Depth Excess* variable on the staging site. 

- **01. GIOPS 10 day Forecast 3D - LatLon**
     - "bathymetry_file_url": "/data/grids/ecc/giops/bathy_meter_giops_latlon_reformatted-ONav.nc",
     - "depthexcess": { "name": "Depth Excess", "unit": "m", "scale": [0, 1000], "equation": "depthexcess([depth], latitude, [votemper] - 273.15, [vosaline], bathy)", "dims": ["time", "latitude", "
longitude"] }
- **04. GIOPS Daily Forecast 3D - LatLon**
     - "bathymetry_file_url": "/data/grids/ecc/giops/bathy_meter_giops_latlon_reformatted-ONav.nc",
     - "depthexcess": { "name": "Depth Excess", "unit": "m", "scale": [0, 1000], "equation": "depthexcess([depth], latitude, [votemper] - 273.15, [vosaline], bathy)", "dims": ["time", "latitude", "longitude"] }
     
- **06. RIOPS Forecast 3D - Polar Stereographic**
     - "bathymetry_file_url": "/data/grids/ecc/riops/bathy_meter_riops_ps5km_reformatted-ONav.nc",
     - "depthexcess": { "name": "Depth Excess", "unit": "m", "scale": [0, 1000], "equation": "depthexcess([depth], latitude, [votemper] - 273.15, [vosaline], bathy)", "dims": ["time", "yc", "xc"] }

**Comment by nsoontie** [comment]
A note on performance: I've noticed for some time now that the Polar Stereographic datasets are much, much slower to load than the lat/lon datasets, especially for derived variables like the acoustic variables and even water velocities.

**Review by nsoontie — APPROVED** [review]

**Comment by dwayne-hart** [comment]
Before merging this in. I would like to hear from the Working Group regarding RIOPS rendering degradation.

**Pull request merged**